purchase_propagate_qty/models/purchase.py

```diff
@@ -0,0 +1,30 @@
+# Copyright 2014-2016 Numérigraphe SARL
+# Copyright 2017 Eficent Business and IT Consulting Services, S.L.
+# Copyright 2021 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import models
+from odoo.tools import float_is_zero
+
+
+class PurchaseOrderLine(models.Model):
+    _inherit = "purchase.order.line"
+
+    def write(self, values):
+        res = super().write(values)
+        if "product_qty" in values or "product_uom" in values:
+            self._propagage_qty_to_moves()
+        return res
+
+    def _propagage_qty_to_moves(self):
+        for line in self:
+            if line.state != "purchase":
+                continue
+            for move in line.move_dest_ids | line.move_ids:
+                if move.state in ("cancel", "done"):
+                    continue
+                if move.product_id != line.product_id:
+                    continue
+                move.product_uom_qty = line.product_uom_qty
+                if float_is_zero(move.product_uom_qty, line.product_uom.rounding):
+                    move._action_cancel()
```

Comment:
This doesn't work when increasing the POL qty.
In purchase stock, a new move is created in such case, with the diff between old POL qty and new POL qty.
This method here sets both moves qties to the new POL qty.
e.g.
POL line with qty set to 100.
User sets the qty to 110.
Odoo creates a new move with qty = 10.
This method sets both qties to 110.